hack/linter.sh

```diff
@@ -7,15 +7,4 @@ then
   exit 0
 fi
 
-DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-
-if ! [ -x "$(command -v golangci-lint)" ]; then
-	echo "Installing GolangCI-Lint"
-	${DIR}/install_golint.sh -b $GOPATH/bin v1.29.0
-fi
-
-export GO111MODULE=on
-golangci-lint run \
-  --timeout 30m \
-  --verbose \
-  --build-tags build
+docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.42.1 golangci-lint run --verbose --deadline 15m0s
```

Comment:
Although a lot of folks won't have docker installed either and therefore lint will break for them?

Reply:
I think we can put a note in the readme that docker needs to be installed for running the linter. 
The current way of installing golangci-lint relies on [godownloader](https://github.com/goreleaser/godownloader) which is not actively maintained anymore. 
Not sure, if we want to spend time writing a script or helper to install golangci-lint on all platforms.
In the end for the lint usecase, docker is just a convenient way to run the linter on all platforms (linux, mac, windows ...), and it also mimics (sort of) what is done in the jx3-pipeline-catalog ...
I can remove the install_golint.sh script and update the readme, wdyt?

Reply:
Thinking about it a bit more, also currently we dont check for versions, we only do checks for the binary, so using docker version mismatch would not be a thing ...

Reply:
Maybe it's best to remove the go lint change from this PR so it can be approved and follow up separately what the best strategy for folks running lint is.